neurolib/optimize/evolution/evolution.py

```diff
@@ -360,14 +399,16 @@ def evalPopulationUsingPypet(self, traj, toolbox, pop, gIdx):
             # store outputs of simulations in population
             pop[idx].fitness.values = fitnessesResult
             # mean fitness value
-            pop[idx].fitness.score = np.nansum(pop[idx].fitness.wvalues) / (len(pop[idx].fitness.wvalues))
+            #pop[idx].fitness.score = np.nansum(pop[idx].fitness.wvalues) / (len(pop[idx].fitness.wvalues))
+            # works with nans and infs:
+            pop[idx].fitness.score = np.ma.masked_invalid(pop[idx].fitness.wvalues).sum() / (len(pop[idx].fitness.wvalues))
```

Comment:
This way the sum works with `np.nan` and `np.inf`.

Reply:
really not needed, but for some reason I do not like `pop[idx].something.something` repeated over and over... I'd do simple class method something like
```python
def _set_population(self, pop, output, fitness, score):
    pop.outputs = output
    pop.fitness.value = fitness
    pop.fitness.score = score

# and actually call it as
self._set_population(pop[idx], returnedOutputs, returnedOutputs, np.ma.masked_invalid(pop[idx].fitness.wvalues).sum() / (len(pop[idx].fitness.wvalues)))
```